### ISSUE TITLE
"GC overhead limit exceeded" caused by liberator.dev/logs

### ISSUE THREAD

**Issue by elenacanovi** [clojure-liberator/liberator#295]
The function 
```
(defn save-log! [id msg]
  (swap! logs #(->> (conj % [id msg])
                    (take log-size))))
```
defined in src/liberator/dev.clj causes a "GC overhead limit exceeded" error,
most probably because, although `take` is lazy, `conj` is not. 
This behavior shows up when using the `wrap-trace` wrapper. 
The error can be reproduced with the minimal code:
```
(loop [] (save-log! 1 2) (recur))
```
Modifying the function as follows:
```
(defn save-log! [id msg]
  (swap! logs #(->> (conj % [id msg])
                    (take log-size)
                    (doall)))
```
is a possible fix.

**Comment by elenacanovi** [comment]
I posted the pull request with the fix shown above (https://github.com/clojure-liberator/liberator/pull/296).

**Comment by ordnungswidrig** [comment]
Fixed in #296
